### PR TITLE
Add -s stagedir option

### PR DESCRIPTION
**Pull request by risner**
If your final dir is a network share, you may wish to copy the file to a staging directory to then have a background thread copy to the network.

This allows me to keep be able to keep up speed. The build directory is a tmpfs, the stage directory is a local hard disk, and the final directory is a NFS share over 1 Gbe network. So it's rather slow.

I had to branch the code, to make other branches. So I'm resubmitting the pull request.

### CODE REVIEW

File: src/chia_plot.cpp [hunk 1]
```diff
@@ -480,6 +503,8 @@ int main(int argc, char** argv)
 	}
 	std::cout << std::endl;
 	std::cout << "Final Directory: " << final_dir << std::endl;
+	if (final_dir != stage_dir)
```
[the remark this answers is not in the document]
Comment (risner) [reply src/chia_plot.cpp]: Changed just now.

File: src/chia_plot.cpp [hunk 2]
```diff
@@ -314,6 +316,9 @@ int main(int argc, char** argv)
 	if(final_dir.empty()) {
 		final_dir = tmp_dir;
 	}
+	if(stage_dir.empty()) {
+		stage_dir = final_dir;
```
[the remark this answers is not in the document]
Comment (risner) [reply src/chia_plot.cpp:320]: I just did a k16 test with every combination. The -s option assumes -d is also set, I need to check if -s is set without -d also set and error out.

-t -2 -s -d
build in -t/-2, write final to -s, then copy to -d (intended function of the code).

-t -2 -d
build in -t/-2, write final to -d (default directout function)

-t -2 -s
build in -t/-2, write final to -s, then it copies to -t
This needs to error out.

[the remark this answers is not in the document]
Comment (risner) [reply src/chia_plot.cpp]: Fixed the default to be tmp_dir.

[the remark this answers is not in the document]
Comment (risner) [reply src/chia_plot.cpp]: Pondering more, just fixing the default to tmp_dir seems to resolve the odd -t -2 -s behavior.

Thanks

[the remark this answers is not in the document]
Comment (madMAx43v3r) [reply src/chia_plot.cpp]: My only goal was that not specifying `stage_dir` results in the same behavior as if this code was never merged. And now it's the case yes.

File: src/chia_plot.cpp [hunk 3]
```diff
@@ -365,6 +370,14 @@ int main(int argc, char** argv)
 		std::cout << "Invalid finaldir: " << final_dir << " (needs trailing '/' or '\\')" << std::endl;
 		return -2;
 	}
+	if(!stage_dir.empty() && stage_dir.find_last_of("/\\") != stage_dir.size() - 1) {
+		std::cout << "Invalid stagedir: " << stage_dir << " (needs trailing '/' or '\\')" << std::endl;
+		return -2;
+	}
+	if(!stage_dir.empty() && tmptoggle) {
```
Comment (madMAx43v3r) [inline src/chia_plot.cpp:377]: This check needs to be above line `stage_dir = tmp_dir;`

Comment (risner) [reply src/chia_plot.cpp]: Moved it above the assignment. Apologies for so many issues, it worked well in my testing. But I apparently didn't check for everything. I appreciate your reviews.

Comment (risner) [reply src/chia_plot.cpp]: Found trailing / check also needed to be moved up.

Comment (madMAx43v3r) [reply src/chia_plot.cpp]: yeah it's normal ;)

**Pull request merged**